### DISCUSSION TITLE
Does Mqttnet run on Linux?

### DISCUSSION THREAD

**Discussion by biapar** [dotnet/MQTTnet#1355] [General]
Does Mqttnet run on Linux?

**Comment by fogzot** [comment]
Yes. We have it in production on Linux servers.

**Comment by SeppPenner** [comment]
I would answer this a bit differently:

Yes, if NetCore 3.1, Net5 or Net6 is concerned.
For the Net Framework and NetStandard, I'm not 100% sure.


I have the latest version of MqttNet (Net6) running as server in Linux (Docker) and Linux-ARM (Docker), so yes it works on Linux 😄 
https://github.com/SeppPenner/SimpleMqttServer

**Reply by chkr1011** [reply]
Do you mean 4.x or 3.x with "latest" version?

**Reply by biapar** [reply]
3.11. I use Net Core 3.1 and I lo e that runs on Linux and Mac. Are there some compatibility problems on Linux?

**Reply by SeppPenner** [reply]
> Do you mean 4.x or 3.x with "latest" version?

3.1.2 or so (I haven't tried 4.x yet).

**Comment by behroozbc** [comment]
Yes. I tested it on a Linux server and Raspberry Pi(arm-32 and arm-64) 
I use those versions of .NET 3.1,5 and 6.
To get more info check those links.
https://blog.behroozbc.ir/basic-mqtt-broker-with-c-sharp
https://blog.behroozbc.ir/start-mqtt-client-with-csharp
https://blog.behroozbc.ir/install-net-on-a-raspberry-pi
